### PR TITLE
Add response to diagnostic

### PR DESCRIPTION
**Pull request by seatuna**

### CODE REVIEW

**Comment by seatuna** [comment]
Comfort: 4
Clarity: 4

Class notes, class repos, Google

File: diagnostic.md [hunk 1]
```diff
@@ -6,21 +6,23 @@ Record your responses inside the fenced code blocks below each question.
     and what are the main task(s) you perform inside an Ember Route?
 
     ```md
-    <!-- your response here -->
+    In the Application Router, you define the routes / URLs for the entire app. The
+    router matches the current URL to the routes you've defined.
+    In an Ember route you can customize specific routes and provide data.
```
Comment (faetea) [inline diagnostic.md:11]: Yes, the Ember Application Router defines the route, and the Ember Route specifies how to load model data using the model hook.

File: diagnostic.md [hunk 2]
```diff
@@ -6,21 +6,23 @@ Record your responses inside the fenced code blocks below each question.
...
     ```
 
 1.  What is the command to generate a route named `boston` nested under
     `campus`?
 
     ```md
-    <!-- your response here -->
+    ember g route campus/boston
```
Comment (faetea) [inline diagnostic.md:18]: yes

File: diagnostic.md [hunk 3]
```diff
@@ -6,21 +6,23 @@ Record your responses inside the fenced code blocks below each question.
...
     ```
 
 1.  Suppose you have a nested route at the URL `/campus/boston`. How would you
     use the `link-to` helper to generate an appropriate link?
 
     ```md
-    <!-- your response here -->
+    {{#link-to 'campus.boston'}}Boston{{/link-to}}
```
Comment (faetea) [inline diagnostic.md:25]: yes

File: diagnostic.md [hunk 4]
```diff
@@ -35,7 +37,12 @@ Record your responses inside the fenced code blocks below each question.
     ```
 
     ```md
-    <!-- your response here -->
+    1. The urls are the same but the file structure is different.  The first route
+       has product nested under products, thus the product folder can be found as a child
+       folder under the products folder.  The second route has products and product as siblings under the application folder.
+    2. When you want to display a 'product' in your browser, the first route will not
+       replace the content from 'products'.  Thus, every time a 'product' is viewed, the
+       content from 'products' will still be visible.  The second route will allow 'product' to replace content from 'products'.
```
Comment (faetea) [inline diagnostic.md:45]: Looks Good.  

Nested route definitions, do require a nested file structure.  This means that the Route and Template files for `'product'` are expected to be found at `app/products/product`.

One of the biggest distinctions is where Ember will render the content.  
The nested `'products'/'product'` instructs Ember to render
`app/products/product/template.hbs` in the `outlet` of the parent `app/products/template.hbs`.  
The non-nested `'product'` instructs Ember to render `app/product/template.hbs` in the context of the application template.

File: diagnostic.md [hunk 5]
```diff
@@ -48,11 +55,12 @@ Record your responses inside the fenced code blocks below each question.
     value `'123'` inside a Route?
 
     ```md
-    <!-- your response here -->
+    When we navigate to /movies/123, the router creates a params object with the
+    movie_id which can be referenced using params.movie_id
```
Comment (faetea) [inline diagnostic.md:59]: yes

File: diagnostic.md [hunk 6]
```diff
@@ -48,11 +55,12 @@ Record your responses inside the fenced code blocks below each question.
...
     ```
 
 1.  Inside a template, how do we reference data provided by a Route?
 
     ```md
-    <!-- your response here -->
+    We use the model method to define a model hook in the route.  The data can then be referenced in the template using dot notation ex. {{model.name}}.
```
Comment (faetea) [inline diagnostic.md:65]: yes

**Comment by faetea** [comment]
6/6

Great Job, you're studying really hard! :+1:

**Pull request closed without merging**